lib/bundler/shared_helpers.rb

```diff
@@ -25,8 +25,18 @@ module Bundler
   module SharedHelpers
     def root
       gemfile = find_gemfile
-      raise GemfileNotFound, "Could not locate Gemfile" unless gemfile
-      Pathname.new(gemfile).untaint.expand_path.parent
+      return Pathname.new(gemfile).untaint.expand_path.parent if gemfile
+
+      bundle_dir = default_bundle_dir
+      raise GemfileNotFound, "Could not locate Gemfile or .bundle/ directory" unless bundle_dir
+
+      bundle_dir.parent
+    end
```

Comment:
Here I'm changing a bit what this method does. I take it that this is not public API, and if any, the public API would be `Bundler.root` (which still works as it worked before).